### PR TITLE
Pal 651 user service redis

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

**Review by pd104923 — DISMISSED** [review]
I've approved the code but I'm curious as to how the h2 db files got added to the repo. These should probably be removed.

File: charts/redis/templates/NOTES.txt [hunk 1]
```diff
@@ -16,48 +16,6 @@
 
 ** Please be patient while the chart is being deployed **
 
-{{- if contains .Values.master.service.type "LoadBalancer" }}
```
Comment (nw1984) [inline charts/redis/templates/NOTES.txt]: Changes in this area are also happening as part of PAL-683

**Review by nw1984 — DISMISSED** [review]
Need to wait for PAL-683 to be committed before committing these changes.

File: user-service/src/main/resources/application-redis.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    redis:
+      timeToLive: 60m
+  # redis.host/port or redis.cluster.nodes configured by k8s
+
+# TODO: spring-boot 2.2.x has cluster bugs, but 2.3.x has health-check bugs
+# Once released, if spring-boot 2.4.x fixes the health-check bugs, re-enable the endpoint here
+management:
+  health:
+    redis:
+      enabled: false
```
Comment (dev930018) [inline user-service/src/main/resources/application-redis.yaml:30]: Should we add this to the backlog?

Comment (developer01189998819991197253) [reply user-service/src/main/resources/application-redis.yaml:30]: @dev930018 whats the expected release date for Spring 2.4? is it in development/predicted?

File: Chart.yaml [hunk 3]
```diff
@@ -29,7 +29,7 @@ dependencies:
   - name: metrics-server
     version: 0.4.0
     repository: file://charts/metrics-server
-    condition: metrics-server.install
```
Comment (ac74475) [inline Chart.yaml]: Why the change? if the change is necessary can we maintain consistency and therefore change the name as well

File: Chart.yaml [hunk 4]
```diff
@@ -42,7 +42,19 @@ dependencies:
...
+    - child: master
+      parent: global.redis.master
+    - child: exports
+      parent: global.redis.exports
   - name: redis-cluster
     version: 0.4.0
-    repository: file://charts/redis-cluster
-    condition: global.redis-cluster.install
+    repository: file://charts/redisCluster
+    condition: global.redisCluster.install
```
Comment (ac74475) [inline Chart.yaml]: same here

File: README.md [hunk 5]
```diff
@@ -112,11 +112,11 @@ Some more important arguments are as follows:
...
 | traefik.install                         | Install the traefik ingress controller, **default=false**
-| metrics-server.install                  | Install the metrics-server to enable horizontal scaling, **default=false**
+| metricsServer.install                   | Install the metrics-server to enable horizontal scaling, **default=false**
 | dashboard.install                       | Install the kubernetes dashboard, **default=false**
-| kafka.install                           | Install Kafka and Zookeeper, **default=true**
-| redis.install                           | Install Redis, **default=true for local, false for aws**
-| redis-cluster.install                   | Install Redis-cluster, **default=false for local, true for aws**
+| global.kafka.install                    | Install Kafka and Zookeeper, **default=true**
+| global.redis.install                    | Install Redis, **default=true for local, false for aws**
+| global.redisCluster.install             | Install Redis-cluster, **default=false for local, true for aws**
```
Comment (ac74475) [inline README.md]: The readme needs to explain the difference between redis and redis cluster

File: charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml [hunk 6]
```diff
@@ -12,14 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{{- if .Values.global.kafka.install }}
```
Comment (ac74475) [inline charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml:15]: surely you only want to install this PV if you are installing Kafka?

File: charts/kafka/templates/kafka-pv.yaml [hunk 7]
```diff
@@ -12,14 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{{- if .Values.global.kafka.install }}
```
Comment (ac74475) [inline charts/kafka/templates/kafka-pv.yaml:15]: again surely we only want to install this PV if kafka is to be installed

File: charts/redis-cluster/values.yaml [hunk 8]
```diff
@@ -45,13 +54,11 @@ image:
...
-##
-# nameOverride:
-
-## String to fully override redis.fullname template
-##
-# fullnameOverride:
+## Busybox init image version
+initImage:
+  repository: busybox
+  tag: latest
```
Comment (ac74475) [inline charts/redis-cluster/values.yaml]: can we use a specific tag

**Review by ac74475 — COMMENTED** [review]
Here are a few initial comments, I think @dev-958 should review this before it gets merged

File: Chart.yaml (hunk 3 above)
Comment (dev930018) [reply Chart.yaml]: In kubernetes some things are treated as DNS names, which dont allow caps but could follow this naming convention. In go templates, they need to follow go variable naming conventions, where `metrics-server` is `metrics subtract server`... The camelCase is better choice in this case i believe (evidenced in the redisCluster vs redis-cluster index and pluck commands)

File: charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml (hunk 6 above)
Comment (dev930018) [reply charts/kafka/charts/zookeeper/templates/zookeeper-pv.yaml:15]: This is already done with the Chart.yaml global.kafka.install condition, it was needed when the pv was part of the root-level templates, but underneath zookeeper templates it's not needed

File: user-service/src/main/resources/application-redis.yaml (hunk 2 above)
Comment (dev930018) [reply user-service/src/main/resources/application-redis.yaml:30]: https://spring.io/blog/2020/06/29/spring-boot-2-4-0-m1-is-now-available
October-November?

Comment (ac74475) [reply user-service/src/main/resources/application-redis.yaml:30]: yes, add a ticket on the backlog

**Review by kg981167224 — APPROVED** [review]

File: user-service/src/main/resources/application-redis.yaml (hunk 2 above)
Comment (dev930018) [reply user-service/src/main/resources/application-redis.yaml:30]: Raised PAL-734 with due-date October

**Review by pd104923 — APPROVED** [review]

**Review by dev-958 — APPROVED** [review]

**Review by ac74475 — APPROVED** [review]

**Pull request merged**